lte/gateway/c/oai/common/log.c

```diff
@@ -1404,6 +1404,9 @@ void log_message_int(
   if (!(g_oai_log.is_async)) {
     _LOG_FREE_ITEM(sync_context_p);
   } else {
+    if (async_context_p == NULL) {
+      return;
+    }
```

Comment:
We wont reach this point if async_context_p was null
Move to right after line 1381

Reply:
This is the claim of Clang-Tidy and the logic seems to hold - unless we make assumptions on immutability of g_oai_log.is_async (which Clang-Tidy is not allowed). 

- Line 1361
  - assume !g_oai_log.is_async
    - So we initialize sync_context_p etc
    - But then lets assume that one of the errors triggers `goto error_event` line 1403
- Line 1403
  - In the time between Line 1361 and 1403, g_oai_log.is_async toggles true
  - We dereference a null async_context_p

Reply:
This is set during task initialization based on the THREAD_SAFE configuration in mme.conf.template. We can assume it is immutable during run time.

Reply:
Our option is to either annotate this as expected behavior to remove the warning - or fix the issue from any future regression - seems to me we might as well put in the future-proof if statement instead of an annotation.

Reply:
Seeing "goto" gives me chills
Fine with moving ahead with this
Maybe change else to else if and add the check there